src/main/java/io/javalin/core/JavalinConfig.java

```diff
@@ -175,7 +175,12 @@ public JavalinConfig configureServletContextHandler(Consumer<ServletContextHandl
         return this;
     }
 
-    public JavalinConfig configureDynamicCompressionStrategy(DynamicCompressionStrategy strategy) {
+    public JavalinConfig setCompressionStrategy(@NotNull DynamicCompressionStrategy strategy) {
+        if(this.inner.server != null) {
+            if(this.inner.server.isRunning()) {
+                return this;
+            }
+        }
```

Comment:
As a rule `set` is not used for setters in Javalin, so this should just be `compressionStrategy` (I would also just remove the if-check).

Reply:
In that case, are we okay with the setter still being able to change compression strategy once the server has been started?

Reply:
> In that case, are we okay with the setter still being able to change compression strategy once the server has been started?

Yes, I really don't mind giving people the freedom to do that, but it shouldn't be an advertised feature. People can set the strategy once on `JavalinConfig` in the `create()` method, but if they want to change it later by playing with the inner config, that's really okay. Sorry about the confusion.

Reply:
No problem, that's what the discussion is for :D